### PR TITLE
User Registration Onboarding Dashboard

### PR DESCRIPTION
**Pull request by stevenelleman**
image

### CODE REVIEW

**Comment by stevenelleman** [comment]
May need to fix two things: (1) dashboard progress bar, (2) a leaderboard details issue I'm running into atm.

**Comment by stevenelleman** [comment]
I need to add: 
- Send encrypted message to connection so they know you registered
- # of onboardings should be at the top of the leaderboard

File: apps/frontend/src/lib/auth/register.ts [hunk 1]
```diff
@@ -207,6 +216,18 @@ export async function applyBackupsToNewUser(password: string): Promise<void> {
           case BackupEntryType.CONNECTION:
             const connection = ConnectionSchema.parse(JSON.parse(backup.backup));
 
+            // Give onboarding credit to first user connection
+            if (!gaveOnboardingCredit) {
+              await updateLeaderboardEntry({
+                authToken: session.authTokenValue,
+                chipIssuer: ChipIssuer.DEVCON_2024, // TODO: make this generic
```
Comment (AndrewCLu) [inline apps/frontend/src/lib/auth/register.ts:223]: should make a note here to update for ETH India

Comment (stevenelleman) [reply apps/frontend/src/lib/auth/register.ts]: I improved this -- I pass in the chipIssuer corresponding to the tap.

File: apps/backend/src/routes/chip/leaderboard_entry.ts [hunk 2]
```diff
@@ -40,15 +40,32 @@ router.post(
...
-      // Update leaderboard entry
-      await controller.UpdateLeaderboardEntry(
-        user.username,
-        chipIssuer,
-        entryType,
-        entryValue
-      );
-
-      return res.status(200).json({});
+      if (entryType === LeaderboardEntryType.USER_REGISTRATION_ONBOARDING) {
```
[the remark this answers is not in the document]
Comment (stevenelleman) [reply apps/backend/src/routes/chip/leaderboard_entry.ts:43]: Added comments in a few places.

**Comment by stevenelleman** [comment]
Retesting and found one minor UI issue. Fixed. Looks good, merging.

**Pull request merged**